plotly_resampler/figure_resampler/figurewidget_resampler.py

```diff
@@ -172,7 +172,7 @@ def _update_spike_ranges(self, layout, *showspikes, force_update=False):
         }
 
         if self._prev_layout is None:
-            return
+            self._prev_layout = layout
```

Comment:
This should allow running in batchmode, see #55

Reply:
@jonasvdd any reason why you wanted to return there? (it could be that I missed smth)

Reply:
aaahn I think this was the base case (Initial relayout call)

Reply:
You could add this code (i.e., setting the `prev_layout` if it is None) in both the `reload_data` and `reset_axes` methods.  
I'll leave the :soccer: in your court! :wink: